test: mamba and micromamba release candidates

jjerphan opened the pull request.

jjerphan commented on repo2docker/buildpacks/conda/install-base-env.bash:
```diff
@@ -5,7 +5,7 @@ set -ex
 cd $(dirname $0)
 
-export MAMBA_VERSION="2.0.5"
+export MAMBA_VERSION="2.0.7.rc0"
```
```suggestion
export MAMBA_VERSION="2.0.7.rc1"
```

jjerphan commented on tests/conda/py35-binder-dir/verify:
```diff
@@ -16,10 +16,10 @@ v = out.split()[1]
 
 out = sh(["micromamba", "--version"])
-assert out == "2.0.5", out
+assert out == "2.0.7.rc0", out
```
```suggestion
assert out == "2.0.7.rc1", out
```

jjerphan commented on tests/conda/py35-binder-dir/verify:
```diff
@@ -16,10 +16,10 @@ v = out.split()[1]
 
 out = sh(["mamba", "--version"])
-assert out == "2.0.5", out
+assert out == "2.0.7.rc0", out
```
```suggestion
assert out == "2.0.7.rc1", out
```

jjerphan commented:
I might reopen this PR for the next releases.

minrk commented:
Looks like a bug that `mamba clean --all` doesn't clean `~/.cache/conda/pkgs/cache/shards`

jjerphan commented:
https://github.com/mamba-org/mamba/pull/4236 should help.

jjerphan commented on repo2docker/buildpacks/conda/install-base-env.bash:
```diff
@@ -5,8 +5,8 @@ set -ex
 
-export MAMBA_VERSION="2.3.3"
-export CONDA_VERSION="25.9.1"
+export MAMBA_VERSION="2.6.0.rc1"
```
```suggestion
export MAMBA_VERSION="2.6.0.rc2"
```

jjerphan commented on tests/conda/py35-binder-dir/verify:
```diff
@@ -16,10 +16,10 @@ v = out.split()[1]
 
 out = sh(["micromamba", "--version"])
-assert out == "2.3.3", out
+assert out == "2.6.0.rc1", out
```
```suggestion
assert out == "2.6.0.rc2", out
```

jjerphan commented on tests/conda/py35-binder-dir/verify:
```diff
@@ -16,10 +16,10 @@ v = out.split()[1]
 
 out = sh(["mamba", "--version"])
-assert out == "2.3.3", out
+assert out == "2.6.0.rc1", out
```
```suggestion
assert out == "2.6.0.rc2", out
```

jjerphan reviewed (commented):
Thank you for your help, @minrk.

Let us use 2.6.0.rc2 once available (see https://github.com/conda-forge/micromamba-feedstock/pull/287).

jjerphan commented on tests/check-tmp:
```diff
@@ -15,7 +15,7 @@ from textwrap import indent
 # directories larger than this are considered a failure
 # a few little files here aren't a problem
-THRESHOLD = 1  # in MB
+THRESHOLD = 7  # in MB
```
@minrk: the cache now hosts a few shards and is now weighs slightly less than 7MB (hence this change). Is it OK for `repo2docker`?

minrk replied:
Our install commands should clean the cache, so needing a change here would indicate that the cache clean commands are not correct, either in mamba or in our invocations.

jjerphan replied:
This might indeed be related to https://github.com/mamba-org/mamba/issues/4247.

jjerphan replied:
Should be resolved by https://github.com/mamba-org/mamba/pull/4249.

jjerphan commented on repo2docker/buildpacks/conda/install-base-env.bash:
```diff
@@ -5,8 +5,8 @@ set -ex
 
-export MAMBA_VERSION="2.3.3"
-export CONDA_VERSION="25.9.1"
+export MAMBA_VERSION="2.6.0.rc3"
```
```suggestion
export MAMBA_VERSION="2.6.0.rc4"
```

jjerphan commented on tests/conda/py35-binder-dir/verify:
```diff
@@ -16,10 +16,10 @@ v = out.split()[1]
 
 out = sh(["micromamba", "--version"])
-assert out == "2.3.3", out
+assert out == "2.6.0.rc3", out
```
```suggestion
assert out == "2.6.0.rc4", out
```

jjerphan commented on tests/conda/py35-binder-dir/verify:
```diff
@@ -16,10 +16,10 @@ v = out.split()[1]
 
 out = sh(["mamba", "--version"])
-assert out == "2.3.3", out
+assert out == "2.6.0.rc3", out
```
```suggestion
assert out == "2.6.0.rc4", out
```

jjerphan commented:
2.6.0 pass. Closing, and I might reopen it later.

jjerphan committed:
test: Use mamba 2.9.0.rc0

Signed-off-by: Julien Jerphanion <git@jjerphan.xyz>

jjerphan committed:
test: Use mamba 2.9.0.rc1

Signed-off-by: Julien Jerphanion <git@jjerphan.xyz>

jjerphan committed:
test: Allow extra keys in micromamba env list --json

micromamba 2.9 includes log_history in JSON output, so assert on envs only.

Signed-off-by: Julien Jerphanion <git@jjerphan.xyz>

jjerphan commented:
2.9.0.rc1 passes all the tests (modulo a relaxation of tests which were too strict).

jjerphan committed:
test: Handle micromamba list --json package wrapper

micromamba 2.9 returns {"packages": [...], "log_history": ...} instead of a
bare list; unwrap packages before asserting.

Signed-off-by: Julien Jerphanion <git@jjerphan.xyz>
